File: Array/codeforces_1879A.cpp

// Problem link ===>> https://codeforces.com/problemset/problem/1879/A
// submission link ===>> https://codeforces.com/contest/1879/submission/241202726

#include <bits/stdc++.h>
#define ll long long
using namespace std;

int main()
{

    ll t;
    cin >> t;

    while (t--)
    {

        ll n;
        cin >> n;

        ll cmp1, cmp2;
        cin >> cmp1 >> cmp2;

        bool possible = true;
        for (ll i = 1; i < n; i++)
        {

            ll u1, u2;
            cin >> u1 >> u2;

            if (u1 >= cmp1 && cmp2 <= u2)
            {
                possible = false;
            }
        }

        if (possible)
        {
            cout << cmp1 << endl;
        }

        else
        {
            cout << -1 << endl;
        }
    }

    return 0;
}

File: Array/codeforces_1879A.py

# Problem link ===>> https://codeforces.com/problemset/problem/1879/A

def solve(n, k):
    if n % k != 0:
        return -1

    groups = n // k
    groups_with_extra = groups // 2 + 1

    if groups_with_extra > k:
        return -1

    scores = [0] * n
    for i in range(0, n, k):
        if i < groups_with_extra * k:
            scores[i:i+k] = list(range(groups - groups_with_extra + 1, -1, -1))
        else:
            scores[i:i+k] = list(range(groups_with_extra, 0, -1))

    return scores

if __name__ == '__main__':
    n = int(input())
    k = int(input())

    scores = solve(n, k)

    if scores == -1:
        print(-1)
    else:
        print(' '.join(map(str, scores)))
